### ISSUE TITLE
Prevent clicking svg links for 'blanking' page

### ISSUE THREAD

**Issue by tecosaur** [org-roam/org-roam#534]
### Description
At the moment, clicking on an `org-protocol://` link in Firefox makes the current page go blank.

#### Steps to Reproduce
I'm using Firefox 75, with a freshly-installed `org-roam` running with Doom Emacs (on Linux), and decktop entry / xdg-mime link set up as per the Appendix.

- Click on a link

#### Expected Results
Upon clicking a cell, that emacs be brought up, and the graph stays.
![image](https://user-images.githubusercontent.com/20903656/80513498-b826f900-89b1-11ea-9ed5-4424a70497a7.png)

#### Actual Results
The graph page goes blank.
![image](https://user-images.githubusercontent.com/20903656/80513540-c8d76f00-89b1-11ea-964b-72d0f0b2816e.png)

I thought adding `target="_blank"` may help, unfortunately while that does stop the link from opening and 'blanking' the current page, it just does it in a new tab.

### Versions
- Emacs: v26.3
- Org-roam commit: 963692f353

p.s. The default graphviz style is both uncomplicated, and underwhelming. How would you feel about setting a prettier default (as above).

**Comment by progfolio** [comment]
> The default graphviz style is both uncomplicated, and underwhelming. How would you feel about setting a prettier default (as above).

I agree that we could do more here, but please open a separate issue to track discussion of this.

**Comment by tecosaur** [comment]
Just wanted to gauge interest on that point first, hence a p.s. :smiley:

**Comment by stites** [comment]
could org-protocol be used here, similar to how the org-roam-protocol bookmarklet works?

**Comment by tecosaur** [comment]
FYI this was fixed for me in https://github.com/tecosaur/emacs-config/commit/3e8cff399e6f3714fa2b9361038bf3857beb71f8

**Comment by progfolio** [comment]
#494 will make this easier to implement. No need for advice, you'll just be able to set the build function directly. I'll take a look at this today and see what I can do.

I notice you're putting a call to `sleep-for` in your build function before writing the html template. Is there a race condition even though call-process is synchronous?

**Comment by tecosaur** [comment]
On nice.

Re: `sleep-for`, without that I found that `f-read-text` returned `""`, so it does look like there's something happening.

**Comment by tecosaur** [comment]
BTW @progfolio if you're interested in extending the graph stuff, you may be interested in https://org-roam.discourse.group/t/application-of-graph-theory-to-roam-link-network/61

**Comment by progfolio** [comment]
Just read through it. Exciting ideas. You bring up some applications which would be very useful as one's graph grows. The performance concerns are well founded, too. As you state, I don't think elisp would be pragmatic for those types of computations, but once we make the graph building flexible enough, separate packages could be developed which can handle these issues. (e.g. distributing/installing a  binary).

**Comment by progfolio** [comment]
@tecosaur: would you mind testing #679? 
Your `org-roam-graph--build-html` function would have to be modified similarly to what I did to `org-roam-graph--build` in that pull. I made it so that the build is executed in a child process and a callback function can be passed. I believe it will obviate the need to sleep in your function as well.
